jcl/src/java.base/share/classes/java/lang/invoke/MethodHandle.java

```diff
@@ -832,33 +833,48 @@ MethodHandle insertArguments(MethodHandle equivalent, MethodHandle unboxingHandl
 	private static final int BSM_OPTIONAL_ARGUMENTS_START_INDEX = 3;
 
 	@SuppressWarnings("unused")
-	private static final MethodHandle resolveInvokeDynamic(Class<?> clazz, String name, String methodDescriptor, long bsmData) throws Throwable {
+	private static final MethodHandle resolveInvokeDynamic(long j9class, String name, String methodDescriptor, long bsmData) throws Throwable {
 		Unsafe unsafe = getUnsafe();
 		MethodHandle result = null;
 		MethodType type = null;
 
 		try {
+			VMLangAccess access = VM.getVMLangAccess();
+			Object internalRamClass = access.createInternalRamClass(j9class);
+			JITHelpers jitHelper = getJITHelpers();
+			Class<?> classObject = null;
+			if (jitHelper.is32Bit()) {
+				classObject = getJITHelpers().getClassFromJ9Class32((int)j9class);
```

Comment:
Don't call `getJITHelpers` again.  It's already been fetched into the `jitHelper` local.  Same on the other branch of the if.